File: frontend/lib/models/intervention.dart
class Intervention {
  final String id;
  final String name;
  final String category;
  final String evidenceRating;
  final List<String> targetDomains;
  final List<String> contraindications;
  final int typicalDurationMinutes;
  final double? costEstimate;

  Intervention({
    required this.id,
    required this.name,
    required this.category,
    required this.evidenceRating,
    required this.targetDomains,
    this.contraindications = const [],
    required this.typicalDurationMinutes,
    this.costEstimate,
  });

  factory Intervention.fromJson(Map<String, dynamic> json) {
    return Intervention(
      id: json['id'],
      name: json['name'],
      category: json['category'],
      evidenceRating: json['evidence_rating'],
      targetDomains: List<String>.from(json['target_domains']),
      contraindications: List<String>.from(json['contraindications'] ?? []),
      typicalDurationMinutes: json['typical_duration_minutes'],
      costEstimate: json['cost_estimate']?.toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'category': category,
      'evidence_rating': evidenceRating,
      'target_domains': targetDomains,
      'contraindications': contraindications,
      'typical_duration_minutes': typicalDurationMinutes,
      'cost_estimate': costEstimate,
    };
  }
}

File: backend/models/intervention.py
"""
Intervention models for the SANA Algorithms Suite.

Contains Pydantic models for CAM interventions and treatments.
"""

from pydantic import BaseModel, Field
from typing import Optional, List
from uuid import UUID, uuid4


class Intervention(BaseModel):
    """Base intervention model."""
    id: UUID = Field(default_factory=uuid4)
    name: str
    category: str  # yoga, meditation, herbal, nutrition, etc.
    evidence_rating: str  # strong, moderate, weak, insufficient
    target_domains: List[str]
    contraindications: List[str] = []
    typical_duration_minutes: int
    cost_estimate: Optional[float] = None


class InterventionDetail(BaseModel):
    """Detailed intervention information."""
    intervention_id: UUID
    description: str
    benefits: List[str] = []
    risks: List[str] = []
    dosage_info: Optional[str] = None
    frequency_recommendation: Optional[str] = None
    research_citations: List[str] = []


class InterventionRecommendation(BaseModel):
    """Model for intervention recommendations."""
    intervention: Intervention
    relevance_score: float
    expected_impact: dict  # domain: expected_improvement
    confidence: float


class InterventionCreate(BaseModel):
    """Model for intervention creation requests."""
    name: str
    category: str
    evidence_rating: str
    target_domains: List[str]
    contraindications: List[str] = []
    typical_duration_minutes: int
    cost_estimate: Optional[float] = None


class InterventionResponse(BaseModel):
    """Model for intervention API responses."""
    id: UUID
    name: str
    category: str
    evidence_rating: str
    target_domains: List[str]
    contraindications: List[str]
    typical_duration_minutes: int
    cost_estimate: Optional[float]

    class Config:
        from_attributes = True
